### PR TITLE
deepin desktop environment: update

### PR DESCRIPTION
**Pull request by wineee**
## Description of changes

# New application

- dde-shell: The taskbar has been refactored using QML.
- dde-grand-search: System-wide desktop search
- deepin-anything: file search tool

https://www.deepin.org/en/deepin-v23-rc-is-officially-released/

image
image

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/deepin-anything-module/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, deepin
+, kernel
+}:
+
+stdenv.mkDerivation {
+  pname = "deepin-anything-module";
+  version = deepin.deepin-anything.version;
+  src = deepin.deepin-anything.src;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/deepin-anything-module/default.nix:9]:
```suggestion
  inherit (deepin.deepin-anything) version src;
```

File: pkgs/desktops/deepin/tools/deepin-anything/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    rev = version;
+    hash = "sha256-OYPsUXMjuU6gG+EzyYl640+2/59n8D5V906CVGwn6Bo=";
+  };
+
+  postPatch = ''
+    substituteInPlace src/CMakeLists.txt \
+      --replace-fail 'add_subdirectory("kernelmod")' " "
+    substituteInPlace src/server/backend/CMakeLists.txt \
+      --replace-warn "/usr" "$out" \
+      --replace-warn "/etc" "$out/etc"
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/tools/deepin-anything/default.nix:30]: Shouldn't this also be --replace-fail?

File: pkgs/desktops/deepin/tools/deepin-anything/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+, pcre
+}:
+
+stdenv.mkDerivation rec {
+  pname = "deepin-anything";
+  version = "6.1.9";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/tools/deepin-anything/default.nix:20]:
```suggestion
    repo = "deepin-anything";
```

File: pkgs/desktops/deepin/library/qt6platform-plugins/default.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+, qt6Packages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "qt6platform-plugins";
+  version = "6.0.16";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/library/qt6platform-plugins/default.nix:19]:
```suggestion
    repo = "qt6platform-plugins";
```

File: pkgs/desktops/deepin/library/qt6integration/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+, gtest
+}:
+
+stdenv.mkDerivation rec {
+  pname = "qt6integration";
+  version = "6.0.16";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/library/qt6integration/default.nix:17]:
```suggestion
    repo = "qt6integration";
```

File: pkgs/desktops/deepin/library/dtkcommon/default.nix [hunk 6]
```diff
@@ -6,13 +6,13 @@
 
 stdenv.mkDerivation rec {
   pname = "dtkcommon";
-  version = "5.6.21";
+  version = "5.6.29";
 
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/library/dtkcommon/default.nix:13]:
```suggestion
    repo = "dtkcommon";
```

File: pkgs/desktops/deepin/go-package/startdde/default.nix [hunk 7]
```diff
@@ -19,13 +19,13 @@
 
 buildGoModule rec {
   pname = "startdde";
-  version = "6.0.13";
+  version = "6.0.14";
 
   src = fetchFromGitHub {
     owner = "linuxdeepin";
     repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/desktops/deepin/go-package/startdde/default.nix:26]:
```suggestion
    repo = "linuxdeepin";
```

File: nixos/modules/services/desktops/deepin/deepin-anything.nix [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        for Linux users, and provides offline search functions.
+      '';
+
+    };
+
+  };
+
+
+  ###### implementation
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktops/deepin/deepin-anything.nix:27]:
```suggestion

```

Those are meaningless copy&pasted comments

File: nixos/modules/services/desktops/deepin/deepin-anything.nix [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+  ###### interface
+
+  options = {
+
+    services.deepin.deepin-anything = {
+
+      enable = lib.mkEnableOption ''
+        anything is derived from everything under Windows, and it is
+        committed to providing a lightning-fast filename search function
+        for Linux users, and provides offline search functions.
```
Comment (SuperSandro2000) [inline nixos/modules/services/desktops/deepin/deepin-anything.nix:18]: That sounds strange when prepending ``Wether to enable anything is derived``

**Review by SuperSandro2000 — COMMENTED** [review]
We shouldn't add new occurences where pname is mistakenly used in fetchFromGitHub's repo argument.

File: nixos/modules/services/desktops/deepin/deepin-anything.nix (hunk 9 above)
Comment (Aleksanaa) [reply nixos/modules/services/desktops/deepin/deepin-anything.nix:18]: Also remove trailing dot

**Comment by wineee** [comment]
@ofborg test deepin

**Review by SuperSandro2000 — APPROVED** [review]
LGTM

**Comment by wineee** [comment]
@ofborg test deepin

**Pull request merged**